packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,487 @@
+import { byteArrayEquals, toHexString } from '@chainsafe/ssz'
+import {
+  Address,
+  MAX_INTEGER,
+  bigIntToHex,
+  bigIntToUnpaddedBuffer,
+  bufferToBigInt,
+  bufferToHex,
+  computeVersionedHash,
+  ecrecover,
+  toBuffer,
+} from '@ethereumjs/util'
+import {
+  computeAggregateKzgProof,
+  freeTrustedSetup,
+  loadTrustedSetup,
+  verifyAggregateKzgProof,
+} from 'c-kzg'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+
+import { BaseTransaction } from './baseTransaction'
+import {
+  BLOB_COMMITMENT_VERSION_KZG,
+  BlobNetworkTransactionWrapper,
+  BlobTransactionType,
+  LIMIT_BLOBS_PER_TX,
+  SignedBlobTransactionType,
+} from './types'
+import { AccessLists, checkMaxInitCodeSize } from './util'
+
+import type {
+  AccessList,
+  AccessListBuffer,
+  AccessListBufferItem,
+  BlobEIP4844TxData,
+  JsonTx,
+  TxOptions,
+  TxValuesArray,
+} from './types'
+import type { Common } from '@ethereumjs/common'
+
+const TRANSACTION_TYPE = 0x05
+const TRANSACTION_TYPE_BUFFER = Buffer.from(TRANSACTION_TYPE.toString(16).padStart(2, '0'), 'hex')
+
+const validateBlobTransactionNetworkWrapper = (
+  versionedHashes: Uint8Array[],
+  blobs: Uint8Array[],
+  commitments: Uint8Array[],
+  kzgProof: Uint8Array
+) => {
+  if (!(versionedHashes.length === blobs.length && blobs.length === commitments.length)) {
+    throw new Error('Number of versionedHashes, blobs, and commitments not all equal')
+  }
+
+  loadTrustedSetup(__dirname + '/kzg/trusted_setup.txt')
+  const verified = verifyAggregateKzgProof(blobs, commitments, kzgProof)
+
+  if (!verified) {
+    throw new Error('KZG proof cannot be verified from blobs/commitments')
+  }
+
+  for (let x = 0; x < versionedHashes.length; x++) {
+    const computedVersionHash = computeVersionedHash(commitments[x])
+    if (!byteArrayEquals(computedVersionHash, versionedHashes[x])) {
+      throw new Error(`commitment for blob at index ${x} does not match versionedHash`)
+    }
+  }
+
+  freeTrustedSetup()
```

Comment:
do we have to load and free trusted setup each time, better would be to load it at the hf activation time somewhere and then never free?

Reply:
we can wrap the c-kzg in a class and have a singleton instantiation of it, just thinking aloud

Reply:
or simply just load it when the client is loaded (and explicitly in tests etc), for the users of the libs, ask them to explicitly load it via docs and examples that we generally provide

Reply:
Maybe.  I'm not going to try and solve this one until we make a final decision on which kzg library we're going to use.  It's definitely not going to be this library in it's current form since it's node only and I want a WASM compiled version.  In all likelihood, we end up using the rust kzg library from Kev and bake the trusted setup right into it.  Once the 4844 ceremony is complete and we have a source of truth for the trusted setup, it should be able to be hardcoded into the library itself so there's no need for this invoking the trusted setup whenever we use the kzg code.

Reply:
hmmm, although i dislike the idea of baking trusted setup into the lib for other chains extensibility but i guess not our problem :shrug:

Reply:
Fair.  In any case, it's not something I'm eager to spend time to solve until I get a better sense of where we're going to land.